R/credentials_gce.R

```diff
@@ -103,7 +103,7 @@ gce_metadata_request <- function(path, stop_on_error = TRUE) {
     path <- substring(path, 2)
   }
   url <- paste0(root_url, "computeMetadata/v1/", path)
-  timeout <- getOption("gargle.gce.timeout", default = 2)
+  timeout <- getOption("gargle.gce.timeout", default = ifelse(!detect_gce(), 0.8, 2))
```

Comment:
This would seem to create a recursive call to `gce_metadata_request()`.

Now that we seem to be converging on the right solution, I plan to make a PR that implements it.

Reply:
@jennybc Alright, thanks :)